src/test/java/org/takes/http/BkBasicTest.java

```diff
@@ -135,7 +134,71 @@ public void exec(final URI home) throws IOException {
     }
 
     /**
+     * BkBasic produces headers with addresses without slashes.
+     *
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void addressesInHeadersAddedWithoutSlashes() throws IOException {
+        final Socket socket = BkBasicTest.createMockSocket();
+        final AtomicReference<Request> ref = new AtomicReference<Request>();
+        new BkBasic(
+            new Take() {
+                @Override
+                public Response act(final Request req) {
+                    ref.set(req);
+                    return new RsEmpty();
+                }
+            }
+        ).accept(socket);
+        final Request request = ref.get();
+        MatcherAssert.assertThat(
+            request,
+            Matchers.is(
+                Matchers.notNullValue()
+            )
+        );
+        MatcherAssert.assertThat(
+            request,
+            Matchers.is(
+                Matchers.instanceOf(
+                    RqWithHeaders.class
+                )
+            )
+        );
+        final Collection<String> head = Lists.newArrayList(request.head());
+        MatcherAssert.assertThat(head, Matchers.not(Matchers.<String>empty()));
+        final Collection<String> found = Collections2.filter(
+            head,
+            new Predicate<String>() {
+                @Override
+                public boolean apply(final String header) {
+                    return header.contains("X-Takes")
+                        && header.contains("Address");
+                }
+            }
+        );
+        for (final String header : found) {
+            MatcherAssert.assertThat(
+                header,
+                Matchers.not(
+                    Matchers.containsString("/")
+                )
+            );
+        }
+        MatcherAssert.assertThat(
```

Comment:
@alexey-krylov  you can extract a header from the request like this
```
  new RqHeaders.Smart(new RqHeaders.Base(request)).single(<header name>)
```

or with default value
```
  new RqHeaders.Smart(new RqHeaders.Base(request)).single(<header name>, <default>)
```